engine/apps/public_api/serializers/escalation_policies.py

```diff
@@ -288,4 +288,5 @@ def update(self, instance, validated_data):
                     instance.num_alerts_in_window = None
                     instance.num_minutes_in_window = None
 
-        return super().update(instance, validated_data)
+        instance.save()
+        return instance
```

Comment:
this is the main focus of this PR. This change fixes things as I _believe_ it's related to an "issue" in `django-rest-framework.serializers.BaseSerializer.update` [here](https://github.com/encode/django-rest-framework/blob/0f39e0124d358b0098261f070175fa8e0359b739/rest_framework/serializers.py#L1049):
```python3
for attr, value in validated_data.items():
  if attr in info.relations and info.relations[attr].to_many:
      m2m_fields.append((attr, value))
  else:
      setattr(instance, attr, value) # <-- here
```

this results in `instance.from_time` being overridden to `type(str)` when it should in fact be a `type(time.time)`. For context the stack trace for the bug I am trying to fix is:
```python3
...
File "/etc/app/apps/public_api/views/escalation_policies.py", line 62, in perform_update
    new_state = serializer.instance.insight_logs_serialized
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/etc/app/apps/alerts/models/escalation_policy.py", line 393, in insight_logs_serialized
    result["from_time"] = self.from_time.isoformat() + " (UTC)"
                          ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'isoformat'
```

From Django's perspective, `from_time` is persisted as a `models.TimeField`.. hence it should _never_ be of `type(str)` when trying to access `self.from_time`.

Reply:
an alternative approach that I _thought_ could work, but doesn't, would be to update [`CustomTimeField.to_internal_value`](https://github.com/grafana/oncall/blob/dev/engine/common/api_helpers/custom_fields.py#L112-L121) to return a `time.struct_time` instead of `str`:
```python3
  def to_internal_value(self, data):
      TIME_FORMAT_LEN = len("00:00:00Z")
      if len(data) == TIME_FORMAT_LEN:
          try:
              time.strptime(data, "%H:%M:%SZ")
          except ValueError:
              raise BadRequest(detail="Invalid time format, should be '00:00:00Z'")
      else:
          raise BadRequest(detail="Invalid time format, should be '00:00:00Z'")
      return data
```
to:
```python3
  def to_internal_value(self, data):
      try:
          return time.strptime(data, "%H:%M:%SZ")
      except ValueError:
          raise BadRequest(detail="Invalid time format, should be '00:00:00Z'")
```

this results in an issue w/ Django trying to persist the value:
```python3
2023-12-18 14:28:26 source=engine:app google_trace_id=none logger=django.request Internal Server Error: /api/v1/escalation_policies/ERRD95A275QUK
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/decorators/csrf.py", line 56, in wrapper_view
    return view_func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/rest_framework/viewsets.py", line 125, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/etc/app/common/api_helpers/mixins.py", line 146, in handle_exception
    return super().handle_exception(exc)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/local/lib/python3.11/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/usr/local/lib/python3.11/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/rest_framework/mixins.py", line 68, in update
    self.perform_update(serializer)
  File "/etc/app/apps/public_api/views/escalation_policies.py", line 61, in perform_update
    serializer.save()
  File "/usr/local/lib/python3.11/site-packages/rest_framework/serializers.py", line 207, in save
    self.instance = self.update(self.instance, validated_data)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/etc/app/apps/public_api/serializers/escalation_policies.py", line 300, in update
    return super().update(instance, validated_data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/etc/app/common/ordered_model/serializer.py", line 41, in update
    return super().update(instance, validated_data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/rest_framework/serializers.py", line 1006, in update
    instance.save()
  File "/etc/app/common/ordered_model/ordered_model.py", line 71, in save
    super().save(*args, **kwargs)
  File "/usr/local/lib/python3.11/site-packages/django/db/models/base.py", line 814, in save
    self.save_base(
  File "/usr/local/lib/python3.11/site-packages/django/db/models/base.py", line 877, in save_base
    updated = self._save_table(
              ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/base.py", line 990, in _save_table
    updated = self._do_update(
              ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/base.py", line 1054, in _do_update
    return filtered._update(values) > 0
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/query.py", line 1231, in _update
    return query.get_compiler(self.db).execute_sql(CURSOR)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 1984, in execute_sql
    cursor = super().execute_sql(result_type)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 1549, in execute_sql
    sql, params = self.as_sql()
                  ^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/mysql/compiler.py", line 57, in as_sql
    update_query, update_params = super().as_sql()
                                  ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 1947, in as_sql
    val = field.get_db_prep_save(val, connection=self.connection)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/fields/__init__.py", line 954, in get_db_prep_save
    return self.get_db_prep_value(value, connection=connection, prepared=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/fields/__init__.py", line 2546, in get_db_prep_value
    value = self.get_prep_value(value)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/fields/__init__.py", line 2541, in get_prep_value
    return self.to_python(value)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/fields/__init__.py", line 2515, in to_python
    parsed = parse_time(value)
             ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/utils/dateparse.py", line 95, in parse_time
    return datetime.time.fromisoformat(value).replace(tzinfo=None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: fromisoformat: argument must be str
```